File: AlphaApps/SmartHome-App-Series/SmartHome-Version-3.0/java/mode/wifi/DeviceAdapter.java
package com.sanjeev.stephan.murmu.ctnetwork.smarthome.mode.wifi;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.sanjeev.stephan.murmu.ctnetwork.smarthome.R;

import java.util.ArrayList;

public class DeviceAdapter extends RecyclerView.Adapter<DeviceViewHolder>
{

    ArrayList<DeviceModel> arrayList;
    DeviceModel device;
    public DeviceAdapter(ArrayList<DeviceModel> arraylist)
    {
        this.arrayList = arraylist;
    }

    public void setDeviceItemClickListener(OnDeviceItemClickListener deviceItemClickListener)
    {
        this.deviceItemClickListener = deviceItemClickListener;
    }
    public OnDeviceItemClickListener deviceItemClickListener;
    public interface OnDeviceItemClickListener
    {
        void onItemClickListener(int position);
        void onStateChangeListener(int position,boolean state);

    }


    @NonNull
    @Override
    public DeviceViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i)
    {
        Context context = viewGroup.getContext();
        LayoutInflater inflater = LayoutInflater.from(context);
        View view = inflater.inflate(R.layout.wifi_device_item_layout, viewGroup, false);
        DeviceViewHolder viewHolder = new DeviceViewHolder(view,deviceItemClickListener);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull DeviceViewHolder viewHolder, int position)
    {
        String link_status ;
        device = arrayList.get(position);
        link_status = "http://" + device.getDeviceIP() + "/"+ device.getDeviceRelay();
        viewHolder.deviceImage.setImageResource(device.getDeviceImage());
        viewHolder.deviceName.setText(device.getDeviceName());
        viewHolder.deviceLocation.setText(device.getDeviceLocation());
        //viewHolder.deviceStatus.setText(link_status);
    }

    @Override
    public int getItemCount() {
        return arrayList.size();
    }
}

File: projects/smarthome/SmartHome-Version-Kotlin/app/src/main/java/com/sanjeev/stephan/murmu/smarthome/sample/DeviceAdapter.kt
package com.sanjeev.stephan.murmu.smarthome.sample

import android.app.Activity
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.sanjeev.stephan.murmu.smarthome.R

class DeviceAdapter(private val context: Activity, private val arrayList: ArrayList<DeviceModel>) :
    RecyclerView.Adapter<DeviceViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): DeviceViewHolder {
        val inflater = LayoutInflater.from(context)
        val view = inflater.inflate(R.layout.item_layout_fragment_devices, parent, false)
        return DeviceViewHolder(view,listenerMeHere)
    }

    override fun getItemCount(): Int {
        return arrayList.size
    }

    override fun onBindViewHolder(holder: DeviceViewHolder, position: Int) {
        val deviceModel: DeviceModel = arrayList.get(position)

        holder.deviceName.text = deviceModel.name
        holder.deviceImage.setImageResource(deviceModel.image)
    }

    lateinit var listenerMeHere : OnMyItemClickListener

    fun setOnMyItemClickListener(listener: OnMyItemClickListener) {
        this.listenerMeHere = listener
    }

    interface OnMyItemClickListener{
        fun onRecyclerItemClickListener(position: Int)
        fun onRecyclerImageClickListener(position: Int)
        fun onRecyclerSwitchToggleListener(position: Int)
    }
}